plugins/nordic-hid/fu-nordic-hid-cfg-channel.c

```diff
@@ -446,6 +446,10 @@ fu_nordic_hid_cfg_channel_add_peer(FuNordicHidCfgChannel *self, guint8 peer_id)
 
 	g_debug("peer 0x%02x discovered", peer_id);
 	fu_device_add_child(FU_DEVICE(self), FU_DEVICE(peer));
+
+	/* Ensure that peripherals connected through a dongle will be updated before the dongle. */
+	fu_device_set_priority(FU_DEVICE(peer), fu_device_get_priority(FU_DEVICE(self)) + 1);
```

Comment:
Does adding FWUPD_DEVICE_FLAG_INSTALL_PARENT_FIRST work instead?

Reply:
We actually want to do the opposite here, so that flag should not be set. btw. From what I observed, the flag modifies the `order`, but not `priority`. 

The `fu_util_update` (called on "update" command) sorts devices based on flags (`FWUPD_DEVICE_FLAG_UPDATABLE` and `FWUPD_DEVICE_FLAG_SUPPORTED`). Apart from this it seems to rely on device order received from `fwupd_client_get_devices` -> `fwupd_client_get_devices_async` -> "GetDevices" daemon call -> `fu_engine_get_devices`.
The `fu_engine_get_devices` sorts device list relying on `priority` and `name`, so that modifying `order` may not help here (unless I missed something).

Reply:
> We actually want to do the opposite here

Reply:
> We actually want to do the opposite here

Isn't that the default? FWUPD_DEVICE_FLAG_INSTALL_PARENT_FIRST is supposed to change the default. I think https://github.com/fwupd/fwupd/commit/9e14761f4387b2d7e1bb0d47b7f2c4098d0257cc might be what you're looking for.

Reply:
I tested with the linked commit, unfortunately it does not seem to help in our use-case.

`nordic-hid` devices use a separate update package per device (each device has it's own `*.cab` file). Would the `order` property help in that case? Is the `order` property used by `fu_util_update`? (I guess I overlooked something here)

Reply:
Ah, that only helps in the case of composite firmware (all in one archive). Order is supposed to be the sorting of devices when there are multiple [different] to choose from, and priority is supposed to be so that fwupd can choose the "best device" when there are multiple possible device baseclasses for the same physical device.

I'm a bit confused how setting the priority changes the order the devices are processed.